### ISSUE TITLE
REPL – No Jupyter Kernels Available, even though i;ve created kernels

### ISSUE THREAD

**Issue by Anna-Pinewood** [zed-industries/zed#21467]
### Check for existing issues

- [X] Completed

### Describe the bug / provide steps to reproduce it

Hello!
## TL;DR
I wanted to try jupyter notebooks run in zed, and i faced the issue: REPL Sessions shows there is no kernels available. i followed [the official guide](https://zed.dev/docs/repl#python), didn't help.

I want to run this as a notebook:
```python
# %%
print("Hello World")
```

## Steps to reproduce
0. I use `uv` packege manager. `uv init`, `uv run hello.py`
```bash
➜  org-agent uv init
Initialized project `org-agent`
➜  org-agent git:(master) ✗ uv run hello.py 
Using CPython 3.10.12 interpreter at: /usr/bin/python3.10
Creating virtual environment at: .venv
Hello from org-agent!
```

1. `source .venv/bin/activate`
2.
```bash
pip install ipykernel
python -m ipykernel install --user
```
3. cmd+shift+P => `repl: refresh kernels`
4. `repl: sessions`

And i see nothing in kernles list: "No Jupyter Kernels Available", even though there is definitely is one in jupyter list
```bash
(org-agent) ➜  org-agent git:(master) ✗ jupyter kernelspec list
Available kernels:
  python3    /home/olipina/.local/share/jupyter/kernels/python3
  ```
<img width="1787" alt="image" src="https://github.com/user-attachments/assets/fb48b2a2-a1cc-42a8-b1e8-0ff6c31c0017">

The kernel installation process via terminal is here:
[cmd_output.txt](https://github.com/user-attachments/files/17995347/cmd_output.txt)

## What else did i try?
1. Specify kernel in project settings.json, also tryed full path, both didn't work
```json
{
    "jupyter": {
        "kernel_selections": {
            "python": "python3"
        }
    }
}
```
2. Create different kernels – in local .venv and in .local.
3. Restart zed to apply changes.

### Environment

Zed: v0.163.3 (Zed)
OS: macOS 14.4.1
Memory: 32 GiB
Architecture: x86_64

However my project is located on remote machine:
```bash
➜  org-agent git:(master) ✗ uname -a
Linux hiplet-20566 5.15.0-82-generic #91-Ubuntu SMP Mon Aug 14 14:14:14 UTC 2023 x86_64 x86_64 x86_64 GNU/Linux
➜  ~ ls .zed_server 
zed-remote-server-stable-0.163.3
```

### If applicable, add mockups / screenshots to help explain present your vision of the feature

<img width="1791" alt="image" src="https://github.com/user-attachments/assets/9da76959-1467-4ab2-9e24-8597b1c558e7">

**Comment by lequangphu** [comment]
I also got this error after moving to uv from pyenv to install and manage python and it's packages.

![Image](https://github.com/user-attachments/assets/5dc559fa-02b4-4d5e-8372-5e3f25dd0b59)

**Comment by giorgioskij** [comment]
I have the same issue. After creating an environment with uv, I still get the "No Jupyter Kernels Available" message.

**Comment by krishpop** [comment]
I have the same issue as well, how do we supply the full path for the kernel spec? Currently this isn't supported even though my selected toolchain environment also contains the path to the site-packages where ipykernel is installed.
